src/vuln_analysis/utils/image_extraction.py

```
@@ -0,0 +1,1028 @@
+        assert crane_proc.stderr is not None
+
+        # Intentionally omit per-process timeouts: image size, registry throughput,
+        # and rootfs write speed have no reliable fixed upper bound. Report progress
+        # while allowing both processes to run to completion.
+        with _ProgressReporter(
+            rootfs_dir,
+            pull_ref,
+            progress=progress,
+            stage=ImageExtractionStage.STREAMING_ROOTFS,
+        ):
+            with concurrent.futures.ThreadPoolExecutor(max_workers=1) as executor:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)